### PR TITLE
Linux C++ test & render tests on CI

### PR DESCRIPTION
**Pull request by louwers**
- I renamed the `core-build` workflow to `linux-ci` and repurposed it.
- Added two jobs to it `render-test` and `test`.
- `render-test` uploads the render test output to S3 when it fails and leaves a comment where you can inspect the HTML output
- Both tests types have some failures, but that is not a reason not to merge this. The tests themselves run fine, except for the fact that xvfb-run is a bit flakey.
- I made inclusion of the Node.js CMake files non-default behind an option. These CMake files download Node.js headers so I don't want to include them by default.

### CODE REVIEW

File: platform/node/DEVELOPING.md [hunk 1]
```diff
@@ -48,13 +48,13 @@ To compile the Node.js bindings and install module dependencies, from the reposi
 #### MacOS
 
 ```bash
-cmake . -B build -G Ninja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=Debug -DMBGL_WITH_COVERAGE=ON
+cmake . -B build -G Ninja -DMLN_WITH_NODE -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=Debug -DMBGL_WITH_COVERAGE=ON
```
Comment (acalcutt) [inline platform/node/DEVELOPING.md:51]: Shouldn't this be -DMLN_WITH_NODE=ON ? same with line 57

Comment (ntadej) [reply platform/node/DEVELOPING.md:51]: I think all should stay `MGL` for now and then we rename all at once. Otherwise it's just confusing.

Comment (louwers) [reply platform/node/DEVELOPING.md:51]: I will keep it, but make a separate PR to rename the other prefixes.

I don't want to introduce new `MBGL_*` options.

Comment (ntadej) [reply platform/node/DEVELOPING.md:51]: Can you also pick these PRs to the `opengl-2` branch (before I setup automatic picking)?

Comment (louwers) [reply platform/node/DEVELOPING.md:51]: Sure.

File: .github/workflows/linux-ci.yml [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      - main
+    paths:
+      - "src/**"
+      - "include/**"
+      - ".github/workflows/linux-ci.yml"
+      - "vendor/**"
+      - "CMakeLists.txt"
+      - metrics/linux-gcc8-release-style.json
+
+permissions:
```
Comment (ntadej) [inline .github/workflows/linux-ci.yml:16]: Where do you use the token? This is not clear why it's needed

File: platform/linux/linux.cmake [hunk 3]
```diff
@@ -137,7 +137,9 @@ target_link_libraries(
 add_subdirectory(${PROJECT_SOURCE_DIR}/bin)
 add_subdirectory(${PROJECT_SOURCE_DIR}/expression-test)
 add_subdirectory(${PROJECT_SOURCE_DIR}/platform/glfw)
```
Comment (ntadej) [inline platform/linux/linux.cmake:139]: Why do we include also `glfw` by default? I think this should also be excluded.

File: platform/node/DEVELOPING.md [hunk 4]
```diff
@@ -48,13 +48,13 @@ To compile the Node.js bindings and install module dependencies, from the reposi
...
 ```bash
-cmake . -B build -G Ninja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=Debug -DMBGL_WITH_COVERAGE=ON
+cmake . -B build -G Ninja -DMLN_WITH_NODE=ON -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=Debug -DMBGL_WITH_COVERAGE=ON
 ```
 
 #### Linux
 
 ```bash
-cmake . -B build -G Ninja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10
+cmake . -B build -G Ninja -DMLN_WITH_NODE -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10
```
Comment (ntadej) [inline platform/node/DEVELOPING.md:57]: Typo - missing `ON`

**Review by ntadej — CHANGES_REQUESTED** [review]

File: .github/workflows/linux-ci.yml (hunk 2 above)
Comment (louwers) [reply .github/workflows/linux-ci.yml:16]: `aws-actions/configure-aws-credentials@v2`

File: platform/linux/linux.cmake (hunk 3 above)
Comment (louwers) [reply platform/linux/linux.cmake:139]: Let's do that in a separate PR?

File: .github/workflows/linux-ci.yml (hunk 2 above)
Comment (louwers) [reply .github/workflows/linux-ci.yml:16]: I added a link

File: platform/linux/linux.cmake (hunk 3 above)
Comment (ntadej) [reply platform/linux/linux.cmake:139]: OK 😊

File: platform/node/DEVELOPING.md (hunk 4 above)
Comment (ntadej) [reply platform/node/DEVELOPING.md:57]: Still missing 😊

**Comment by louwers** [comment]
I fixed the merge conflict.

@ntadej I will fix change the `MBGL_` prefix in a separate PR. Do you think it is OK to merge?

File: .github/workflows/linux-ci.yml [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+            libcurl4-openssl-dev \
+            libuv1-dev \
+            libjpeg-dev \
+            libpng-dev \
+            libglfw3-dev
+
+      - name: Build Maplibre GL Core
+        run: |
+          cmake --version
+          cmake -B build -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DMBGL_WITH_COVERAGE=ON
```
Comment (ntadej) [inline .github/workflows/linux-ci.yml:56]: You should recover Ninja and clang-tidy I guess?

File: platform/node/DEVELOPING.md (hunk 4 above)
Comment (ntadej) [reply platform/node/DEVELOPING.md:57]: Readme still not correct!

**Review by ntadej — DISMISSED** [review]
Unfortunately you reverted some changes when solving conflicts. Also you keep resolving one typo which you do not fix 😉

File: .github/workflows/linux-ci.yml (hunk 5 above)
Comment (louwers) [reply .github/workflows/linux-ci.yml:56]: Oops. I did that, but something went wrong during the rebase. 🤦‍♂️

Comment (louwers) [reply .github/workflows/linux-ci.yml]: Thanks for catching this. I added it back.

**Comment by louwers** [comment]
@ntadej @ovivoda OK to merge?

Test & render-test fail but fixing them is out of scope for this PR.

**Comment by ntadej** [comment]
> @ntadej @ovivoda OK to merge? I may have to tweak uploading the render test results a bit, right now forks don’t have the right permissions.
> 
> Test & render-test fail but fixing them is out of scope for this PR.

Then disable them. The CI should not be red 😊

**Comment by louwers** [comment]
> > @ntadej @ovivoda OK to merge? I may have to tweak uploading the render test results a bit, right now forks don’t have the right permissions.
> > Test & render-test fail but fixing them is out of scope for this PR.
> 
> Then disable them. The CI should not be red 😊

I disagree. CI is there to indicate a problem. We need to enable CI to make the problem apparent.

The 'build' step is green, the two steps that this PR add are red.

**Comment by ntadej** [comment]
Yes, but how does the author of another PR know? Also then we need to force-merge every PR.

**Comment by louwers** [comment]
@ntadej We don't need to force merge because requiring CI to be green is not a requirement for merging.

People can see the history of workflows and see that they are already failing on `main`. It is not ideal and I agree we need to fix these problems ASAP, but my reasoning is if these checks were already part of the CI we would not disable them just because they are red.

**Comment by louwers** [comment]
@ntadej There is a team of engineers working on the core... I would not worry about CI staying red too long... They are actively resolving the problems that appear. The reason that the render tests fail for example is that we did not catch a regression in a recent PR.

**Comment by ntadej** [comment]
I leave it up to Ovi.

**Comment by louwers** [comment]
> I leave it up to Ovi.

Alright! Thanks for your critical eye. 🙂

**Comment by ovivoda** [comment]
@louwers @ntadej Nice to read the debate here! I think we should merge this but also open an issue in the repo to address the missing parts. 

I was never a fan of big PR's.

**Review by ovivoda — APPROVED** [review]

**Pull request merged**